src/NUnitFramework/framework/Internal/PreFilter.cs

```diff
@@ -0,0 +1,208 @@
+            foreach (FilterElement filter in _filters)
+            {
+                if (type.FullName == filter.ClassName)
+                {
+                    ++filterCount;
+                    if (method.Name == filter.MethodName)
+                        return true;
+                }
+            }
+
+            // If no match, we succeed only if there were no method filters for this fixture
+            return filterCount == 0;
+        }
+
+        #region Nested FilterElementType Enumeration
+
+        private enum FilterElementType
+        {
+            Unknown,
+            Namespace,
+            Fixture,
+            Method
+        }
+
+        #endregion
+
+        #region Nested FilterElement Class
+
+        private class FilterElement
+        {
+            public FilterElementType ElementType = FilterElementType.Unknown;
+            public string Text;
+            public string ClassName;
+            public string MethodName;
+
+            public FilterElement(string text)
+            {
+                int argStart = text.IndexOfAny(ARG_START);
+                Text = argStart > 0
+                    ? text.Substring(0, argStart)
+                    : text;
+
+                // Filter may be the fullname of a test method, if so extract class name
+                int lastdot = Text.LastIndexOf('.');
+                if (lastdot > 0)
+                {
+                    ClassName = Text.Substring(0, lastdot);
+                    MethodName = Text.Substring(lastdot + 1);
+                }
+            }
+
+            public bool Match(Type type)
+            {
+                switch(ElementType)
+                {
+                    default:
+                    case FilterElementType.Unknown:
+                        return MatchUnknownElement(type);
+                    case FilterElementType.Fixture:
+                        return MatchFixtureElement(type);
+                    case FilterElementType.Namespace:
+                        return MatchNamespaceElement(type);
+                    case FilterElementType.Method:
+                        return MatchMethodElement(type);
+                }
+            }
+
+            private bool MatchUnknownElement(Type type)
+            {
+                if (MatchFixtureElement(type))
+                {
+                    ElementType = FilterElementType.Fixture;
+                    return true;
+                }
+
+                if (MatchNamespaceElement(type))
+                {
+                    ElementType = FilterElementType.Namespace;
+                    return true;
+                }
+
+                if (MatchMethodElement(type))
+                {
+                    ElementType = FilterElementType.Method;
+                    return true;
+                }
+
+                return false;
+            }
+
+            private bool MatchFixtureElement(Type type)
+            {
+                return type.FullName == Text;
```

Comment:
What about nested fixtures? The containing class would be treated as a namespace. I'd expect the fixture with Type.FullName `Namespace.MyClass+MyNestedFixture` to be included if I specify the filter `Namespace.MyClass`.

Reply:
Try this with `--test`. I don't believe it works that way. NUnit has never treated nested classes as nested. That is, we do not consider the class in which it's nested to be a "parent" in any sense. That's why the V2 gui lists `Namespace.MyClass` and `Namespace.MyClass+MyNestedFixture` as entirely separate entries. 

We've had a rather big feature in our list for years to __start__ treating nested classes in this way, but we've never taken it on.

OTOH If I'm wrong and `--test` works the way you say, then we should make `--prefilter` work that way as well.

Reply:
You're correct; NUnit does not treat containing types as namespaces.